libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1034,24 +1034,32 @@ void AP_GPS::Write_DataFlash_Log_Startup_messages()
 
 /*
   return the expected lag (in seconds) in the position and velocity readings from the gps
+  return true if the GPS hardware configuration is known or the delay parameter has been set
  */
-float AP_GPS::get_lag(uint8_t instance) const
+bool AP_GPS::get_lag(uint8_t instance, float &lag_sec) const
 {
     // return lag of blended GPS
-    if (instance == GPS_MAX_RECEIVERS) {
-        return _blended_lag_sec;
+    if (instance == GPS_BLENDED_INSTANCE) {
+        lag_sec = _blended_lag_sec;
+        // auto switching uses all GPS receivers, so all must be configured
+        return all_configured();
     }
 
     if (_delay_ms[instance] > 0) {
         // if the user has specified a non zero time delay, always return that value
-        return 0.001f * (float)_delay_ms[instance];
+        lag_sec = 0.001f * (float)_delay_ms[instance];
+        // the user is always right !!
+        return true;
     } else if (drivers[instance] == nullptr || state[instance].status == NO_GPS) {
-        // no GPS was detected in this instance
-        // so return a default delay of 1 measurement interval
-        return 0.001f * (float)get_rate_ms(instance);
+        // no GPS was detected in this instance so return a default delay of 1 measurement interval
+        lag_sec = 0.001f * (float)get_rate_ms(instance);
+        // check lack of GPS is consistent with user expectation
+        return state[instance].status == NO_GPS;
```

Comment:
This is a misunderstanding of how the GPS library works. A driver being nullptr will result in either a new driver on the first tick of startup, otherwise the status will always be set to NO_GPS. https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_GPS/AP_GPS.cpp#L448 a null driver instance and NO_GPS are meant to always be synonmous and the code attempts to enforce this https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_GPS/AP_GPS.cpp#L614

So as far as I can tell every time you eve actually call this code it will always return true on this particular line. Which may be the desired behaviour, but I can't see any scenario where the check will ever not be true.